File: abc/abc129/abc129_b.cs
// https://atcoder.jp/contests/abc129/submissions/8303193

using System;
using System.Linq;

namespace CSharp
{
    class Program
    {
        public static void Main()
        {
            var N = int.Parse(Console.ReadLine());
            var W = Console.ReadLine().Split(' ').Select(int.Parse).ToArray();

            var l = W.Sum();
            var r = 0;
            var temp = l - r;
            for (int i = N - 1; i >= 0; i--)
            {
                l -= W[i];
                r += W[i];
                if (temp < Math.Abs(l - r))
                {
                    break;
                }
                else
                {
                    temp = Math.Abs(l - r);
                }
            }
            Console.WriteLine(temp);
        }
    }
}

File: abc/abc129/abc129_b.fs
// https://atcoder.jp/contests/abc129/submissions/12152057

let N = stdin.ReadLine() |> int
let W = stdin.ReadLine().Split() |> Array.map int

let rec solver l r t i =
    if i = -1 then t
    else
        let ln = l - W.[i]
        let rn = r + W.[i]
        if t < abs (ln - rn) then t
        else solver ln rn (abs (ln - rn)) (i - 1)

solver (Array.sum W) 0 (Array.sum W) (N - 1) |> stdout.WriteLine
